service/resource/cloudformation/current.go

```diff
@@ -1,7 +1,44 @@
 package cloudformation
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsCF "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for AWS stack")
+
+	stackName := key.MainStackName(customObject)
+
+	describeInput := &awsCF.DescribeStacksInput{
+		StackName: aws.String(stackName),
+	}
+	describeOutput, err := r.awsClient.DescribeStacks(describeInput)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if len(describeOutput.Stacks) != 1 {
```

Comment:
My understanding is GetCurrentState should fall through if the resource doesn't exist rather than erroring.

See https://github.com/giantswarm/kvm-operator/blob/7bf46cd0c5130d104516c854c420d78b99117eef/service/resource/deployment/current.go#L30

If more than 1 stack is returned I think its good to error but IMO should be a notFound error.

Reply:
In case there is no stack the returned state should be empty. This knowledge is then used to decide that the desired state should be used as create state to actually create the stack when applying the computed patch/change.

Reply:
Done, now when the stack is not found (checking the error string returned, aws api and sdk don't help here) an empty stack and a nil error are returned (not sure about the latter, could you please confirm?) and if more than 1 is found, the notFoundError is returned, pls have another look.

Reply:
Yes the AWS API is horrible here :( I think we should add an error matcher to error.go. See some examples from the old resources.

https://github.com/giantswarm/aws-operator/blob/e796869c17eb3a40322c3197463adf529b8d5cdd/resources/aws/errors.go